src/server/middleware/prometheus.ts

```diff
@@ -0,0 +1,54 @@
+import fastify, {
+  FastifyInstance,
+  FastifyReply,
+  FastifyRequest,
+} from "fastify";
+import { env } from "../../utils/env";
+import { enginePromRegister, recordMetrics } from "../../utils/prometheus";
+
+export const withPrometheus = async (server: FastifyInstance) => {
+  if (!env.METRICS_ENABLED) {
+    console.log("Metrics are disabled");
```

Comment:
Let's use the `logger` helper so all logging is consolidated and reuses the same pattern/libraries.

Reply:
this log line slipped out when debugging, do we even need it?

Reply:
Not really, removing is fine too.